### PR TITLE
tests/gnrc_netif: enable CI tests

### PR DESCRIPTION
**Pull request by cladmi**
### Contribution description

Enable automated testing.

### Issues/PRs references

Goal is to have it for the release.

### CODE REVIEW

**Comment by miri64** [comment]
There was a regression introduced in #8215 that causes this test to crash. I will try to find a solution.

**Comment by miri64** [comment]
The problem is caused, due to IPv6 packets having multiple subscribers due to the `gnrc_pktdump` subscription in this test. This special edge-case wasn't tested for #8215 and leads to some weird handling in the `users` setting of the `pktsnip`s, which I currently try to wrap my head around

**Comment by miri64** [comment]
Regression fixed in #9672. I will also provide other fixes.

**Comment by miri64** [comment]
> I will also provide other fixes.

With https://github.com/RIOT-OS/RIOT/pull/9675 it work at least on `native` again. The "fix" @cladmi proposed in #9007, did not work for me locally on `samr21-xpro` (even with sleeping for 1 sec).

File: tests/gnrc_netif/Makefile [hunk 1]
```diff
@@ -27,6 +27,8 @@ CFLAGS += -DGNRC_NETIF_GROUPS_NUMOF=8
 CFLAGS += -DLOG_LEVEL=LOG_NONE
 CFLAGS += -DTEST_SUITES
 
+TEST_ON_CI_WHITELIST += all
```
Comment (miri64) [inline tests/gnrc_netif/Makefile]: Maybe, so we get around the synchronization issue for now use `native` instead of `all`? :stuck_out_tongue:

Comment (cladmi) [reply tests/gnrc_netif/Makefile]: With your https://github.com/RIOT-OS/RIOT/pull/9672 PR I do not have any synchronization issue anymore on all the boards I tested. https://github.com/RIOT-OS/RIOT/pull/9675#issuecomment-409729428

Comment (cladmi) [reply tests/gnrc_netif/Makefile]: I merged your PR to test with murdock.

Comment (miri64) [reply tests/gnrc_netif/Makefile]: Interesting... I wasn't able to on IoT-LAB's `samr21-xpro`. Let me retry with a local `samr21-xpro`

**Comment by miri64** [comment]
Ok... @cladmi and I retested on several different `samr21-xpro` and came to inconclusive results :-/.

**Comment by miri64** [comment]
#9672 was merged. Please rebase.

**Comment by miri64** [comment]
(bumping up a release-related PR ;-))

**Comment by miri64** [comment]
So how do we proceed on this? Do we just hope, that the `samr21-xpro`s connected to Murdock are not in the set of failing once? Do we fix the problem? The current build seemed to have been without problems.

**Comment by cladmi** [comment]
I would try fixing it for the release, but not for RC2.

**Comment by miri64** [comment]
Depends on #9719

**Comment by miri64** [comment]
@kaspar030 the false negative tests are all produced by a single worker so far (`pi-0979730f`). Is it safe to just unplug it to see what's wrong with it?

**Comment by kaspar030** [comment]
yes, feel free!

**Comment by miri64** [comment]
Please provide a backport.

**Comment by cladmi** [comment]
Backport provided in #9736
